### PR TITLE
Health log book

### LINKED ISSUE #823: Health Log Book
I am Anurag Mukherjee, I am a participant of GSSoC'21
# Aim

Keeping daily records of a person's food and workout

# Details
The script will contain the daily records of food and workouts of a person. The user will only enter the name of the foods, it will be stored in a text file with corresponding date and time. Similarly, the user will enter the types of workout at their workout time. The workouts will be stored in a text file with date and time. The user can also retrieve the data whenever they want to see.

## Advantages of the script:-

This script will help the user to maintain his/her health, as it is keeping a daily track of their diet and workout style. At the end of the day, or at the end of the week, he/she can analyze the data and can lead a healthy and fit lifestyle.

## Modules will be used:-

1)  datetime 

I will write clean and easy to understand code. Please assign this issue to me.

### PR DESCRIPTION
**Pull request by anuragmukherjee2001**
# Description
The script will contain the daily records of food and workouts of a person. The user will only enter the name of the foods, it will be stored in a text file with corresponding date and time. Similarly, the user will enter the types of workout at their workout time. The workouts will be stored in a text file with date and time. The user can also retrieve the data whenever they want to see.

## Modules Used:-
1) datetime

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [ ] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [ ] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

File: Health_Log_Book/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+### Advantages of the script:-
+This script will help the user to maintain his/her health, as it is keeping a daily track of their diet and workout style. At the end of the day, or at the end of the week, he/she can analyze the data and can lead a healthy and fit lifestyle.
+
+### Modules Used:-
+
+1) datetime
+
+### Images of the script:-
+<img src = "https://user-images.githubusercontent.com/62845847/113548479-3998f980-960d-11eb-806b-4321cec05a98.png" width = "400" height = "300">
```
Comment (kaustubhgupta) [inline Health_Log_Book/README.md:14]: This is not the complete image of the script and therefore you can remove this image. Remove this section heading too.

File: Health_Log_Book/main.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# It is a health management system
```
Comment (kaustubhgupta) [inline Health_Log_Book/main.py:1]: Remove these 4 line comment

File: Health_Log_Book/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+The script will contain the daily records of food and workouts of a person. The user will only enter the name of the foods, it will be stored in a text file with corresponding date and time. Similarly, the user will enter the types of workout at their workout time. The workouts will be stored in a text file with date and time. The user can also retrieve the data whenever they want to see.
+
+### Advantages of the script:-
+This script will help the user to maintain his/her health, as it is keeping a daily track of their diet and workout style. At the end of the day, or at the end of the week, he/she can analyze the data and can lead a healthy and fit lifestyle.
+
+### Modules Used:-
+
+1) datetime
+
+### Images of the script:-
```
Comment (kaustubhgupta) [inline Health_Log_Book/README.md:13]: As this is not the complete script screenshot, you can remove this image. Remove the section heading too

File: Health_Log_Book/main.py (hunk 2 above)
Comment (kaustubhgupta) [inline Health_Log_Book/main.py:1]: Remove these 4 lines and unnecessary comments from the program

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Create a GUI application for this. The current code is not up to the mark for level 2. Also, instead of text files, use mysqlite to store details. If you don't have any experience for that then we need to decrease the level. 

Thoughts? @antrikshmisri @santushtisharma10

**Comment by antrikshmisri** [comment]
> Create a GUI application for this. The current code is not up to the mark for level 2. Also, instead of text files, use mysqlite to store details. If you don't have any experience for that then we need to decrease the level.
> 
> Thoughts? @antrikshmisri @santushtisharma10

Agreed.

**Comment by anuragmukherjee2001** [comment]
Ok sir, I will try to do all the suggested changes. As lots of changes are suggested, so it may take some time. And 1 question sir, can I use SQLite??
@antrikshmisri @santushtisharma10 @kaustubhgupta

**Comment by santushtisharma10** [comment]
> Ok sir, I will try to do all the suggested changes. As lots of changes are suggested, so it may take some time. And 1 question sir, can I use SQLite??
> @antrikshmisri @santushtisharma10 @kaustubhgupta

Yes

**Comment by kaustubhgupta** [comment]
@anuragmukherjee2001 updates on this?

**Comment by anuragmukherjee2001** [comment]
> @anuragmukherjee2001 updates on this?

Yes sir, I am making the suggested changes. I will push the changes within tomorrow.

**Comment by anuragmukherjee2001** [comment]
I updated the code. Please check it sir.

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
If I click on show exercise/food log without entering any information, an exception is raised:
```python
Exception in Tkinter callback
Traceback (most recent call last):
  File "C:\Users\Lenovo\anaconda3\lib\tkinter\__init__.py", line 1705, in __call__
    return self.func(*args)
  File "d:/Projects/Amazing-Python-Scripts/Health_Log_Book/main.py", line 54, in show_food
    cor.execute('''SELECT * from food''')
sqlite3.OperationalError: no such table: food
``` 
Handle this case by giving popup that "No information available"

Also, on clicking show log, the information is printed in the terminal. Display that information in the application GUI only. Create a text area called results to show them

**Comment by anuragmukherjee2001** [comment]
Ok sir, trying to correct the mistakes

**Comment by kaustubhgupta** [comment]
@anuragmukherjee2001 updates?

**Comment by anuragmukherjee2001** [comment]
> @anuragmukherjee2001 updates?

I corrected the case which was giving an error. I am working on the feature of showing the information on GUI.

**Comment by anuragmukherjee2001** [comment]
I have made all the changes which were suggested. Please Check the PR.

**Review by kaustubhgupta — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM

**Pull request merged**